Build failures on armhfp architecture

This is low priority, but maybe it's an easy fix? 

In building cherrytree-future on armhfp for Fedora 31 and Fedora 32, the build fails at the same place. This only happens on armhfp, not x86_64 or aarch64. I'm guessing this is the first place that a 64-bit  variable is used, since armhfp is a 32-bit architecture.

```[ 93%] Building CXX object src/ct/CMakeFiles/cherrytree_shared.dir/ct_misc_utils.cc.o
/builddir/build/BUILD/cherrytree-26afcc6b27635d5b710c481daf9ae74c8ab20ea8/future/src/ct/ct_misc_utils.cc: In function 'std::string str::time_format(const string&, const gint64&)':
/builddir/build/BUILD/cherrytree-26afcc6b27635d5b710c481daf9ae74c8ab20ea8/future/src/ct/ct_misc_utils.cc:740:41: error: cannot convert 'const gint64*' {aka 'const long long int*'} to 'const time_t*' {aka 'const long int*'}
  740 |     std::tm* localtime = std::localtime(&time);
      |                                         ^~~~~
      |                                         |
      |                                         const gint64* {aka const long long int*}
In file included from /usr/include/glib-2.0/glib/gtypes.h:35,
                 from /usr/include/glib-2.0/glib/galloca.h:32,
                 from /usr/include/glib-2.0/glib.h:30,
                 from /usr/include/glibmm-2.4/glibmm/unicode.h:23,
                 from /usr/include/glibmm-2.4/glibmm/ustring.h:21,
                 from /usr/include/pangomm-1.4/pangomm/attrlist.h:6,
                 from /usr/include/pangomm-1.4/pangomm.h:55,
                 from /builddir/build/BUILD/cherrytree-26afcc6b27635d5b710c481daf9ae74c8ab20ea8/future/src/ct/ct_misc_utils.cc:24:
/usr/include/time.h:123:44: note:   initializing argument 1 of 'tm* localtime(const time_t*)'
  123 | extern struct tm *localtime (const time_t *__timer) __THROW;
      |                              ~~~~~~~~~~~~~~^~~~~~~
make[2]: *** [src/ct/CMakeFiles/cherrytree_shared.dir/build.make:356: src/ct/CMakeFiles/cherrytree_shared.dir/ct_misc_utils.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:242: src/ct/CMakeFiles/cherrytree_shared.dir/all] Error 2
make: *** [Makefile:183: all] Error 2

--- a/future/CMakeLists.txt
+++ b/future/CMakeLists.txt
@@ -58,7 +58,12 @@ endif()
 # Create the configuration files config.h in the root dir
 configure_file(${CMAKE_SOURCE_DIR}/config.h.cmake ${CMAKE_SOURCE_DIR}/config.h)
 
-
+option(BUILD_TESTING "Build tests" ON)
+if (BUILD_TESTING)
+  message("Tests are ON")
+else()
+  message("Tests are OFF")
+endif()
 
 include(FindPkgConfig)
 pkg_check_modules(GTKMM gtkmm-3.0 REQUIRED)
@@ -68,7 +73,9 @@ pkg_check_modules(PANGOMM pangomm-1.4 REQUIRED)
 pkg_check_modules(GSPELL gspell-1 REQUIRED)
 pkg_check_modules(LIBXML libxml++-2.6 REQUIRED)
 pkg_check_modules(SQLITE sqlite3 REQUIRED)
-pkg_check_modules(CPPUTEST cpputest  REQUIRED)
+if(BUILD_TESTING)
+  pkg_check_modules(CPPUTEST cpputest  REQUIRED)
+endif()
 
 add_subdirectory(src/7za)
 
@@ -89,8 +96,10 @@ include_directories(
 
 add_subdirectory(src/ct)
 
-enable_testing()
-add_subdirectory(tests)
+if(BUILD_TESTING)
+  enable_testing()
+  add_subdirectory(tests)
+endif()
 
 
 

--- a/future/src/ct/ct_misc_utils.cc
+++ b/future/src/ct/ct_misc_utils.cc
@@ -735,7 +735,7 @@ std::string str::re_escape(const std::string& text)
     return Glib::Regex::escape_string(text);
 }
 
-std::string str::time_format(const std::string& format, const gint64& time)
+std::string str::time_format(const std::string& format, const time_t& time)
 {
     std::tm* localtime = std::localtime(&time);
     char buffer[100];

--- a/future/src/ct/ct_misc_utils.h
+++ b/future/src/ct/ct_misc_utils.h
@@ -212,7 +212,7 @@ std::string xml_escape(const std::string& text);
 
 std::string re_escape(const std::string& text);
 
-std::string time_format(const std::string& format, const gint64& time);
+std::string time_format(const std::string& format, const time_t& time);
 
 int symb_pos_to_byte_pos(const Glib::ustring& text, int symb_pos);
 int byte_pos_to_symb_pos(const Glib::ustring& text, int byte_pos);

